### DISCUSSION TITLE
Flaky test: `TestGoExtendCore` - `cannot define methods on objects from outside this module`

### DISCUSSION THREAD

**Discussion by sipsma** [dagger/dagger#12283] [maintainers]
Started seeing this one a bunch recently:
* Cloud trace: https://dagger.cloud/dagger/traces/ce4f383ede89e9bf39d00fe89aacd444?span=087b98784bf05440#909ebe7a106b977f
* GHA job: https://github.com/dagger/dagger/actions/runs/9998434018/job/27637312803?pr=7978

From what I can tell looking in the engine logs above it, it *does* appear the expected error is likely getting hit, it's just not getting relayed to the clients as expected.

cc @vito @jedevc no initial clues what changes would have resulted in this (if any)

**Comment by sipsma** [comment]
@vito mentioned this should be fixed by otel draining re-work: https://github.com/dagger/dagger/pull/7989#issuecomment-2239244084

**Comment by jedevc** [comment]
Going through old CI issues, I can't recall having seen this one at all recently.

Guess the OTEL rework has been successful :tada:

**Comment by cwlbraa** [comment]
[pretty sure this is still happening](https://dagger.cloud/dagger/traces/7b25518867c0f2d2531570db190bc37e#L580-a86f3bb556663896)
